client/src/game/shapes/utils.ts

```diff
@@ -189,10 +191,14 @@ export function deleteShapes(shapes: readonly IShape[], sync: SyncMode): void {
         const sel = shapes[i]!;
         if (sync !== SyncMode.NO_SYNC && !accessSystem.hasAccessTo(sel.id, false, { edit: true })) continue;
         const gId = getGlobalId(sel.id);
-        if (gId) removed.push(gId);
+        if (gId) {
+            removed.push(gId);
+            noteSystem.unhookShape(sel.id);
```

Comment:
In the future these fixes to other things, should be done in separate PRs ideally.

Reply:
Yeah, sorry. I can revert and include it as a separate PR if you want.